chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -73,7 +73,9 @@ object Vec {
 
     def doConnect(sink: T, source: T) = {
       // TODO: this looks bad, and should feel bad. Replace with a better abstraction.
-      val hasDirectioned = vec.sample_element match {
+      // NOTE: Must use elts.head instead of vec.sample_element because vec.sample_element has
+      //       WireBinding which does not have a direction
+      val hasDirectioned = elts.head match {
```

Comment:
This method already requires that elts are nonEmpty.

It's not really clear what Vec(Seq.empty) should mean although I do think constructing a Vec[Nothing] such that so long as the user does not try to index the Vec, it's fine for generators to do `Vec.fill(0){...}`

As for elts already being bound to a Wire, I guess this does mean you couldn't make multi-dimensional Vecs of Module IO. I think we should continue discussion #571 for a complete solution. In the meantime, I believe that this PR fixes a really big problem even if it doesn't completely solve the fundamental issue.